sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -1423,18 +1438,46 @@ public void processElement(ProcessContext c) {
     }
 
     @Setup
-    public void setup() throws Exception {
-      // set up non-serializable values here.
-      spannerAccessor = SpannerAccessor.create(spannerConfig);
+    public void setup() {
+      spannerAccessor = spannerAccessors.get(spannerConfig);
+      if (spannerAccessor == null) {
+        synchronized (spannerAccessors) {
```

Comment:
I guess this could could be common code in SpannerAccessor(may be part of SpannerAccessor.create()) and could be shared as SpannerAccessor is called from other places as well.

Reply:
as above, moved refcounting into SpannerAccessor.